### ISSUE TITLE
[cob_twist_controller] refactor constraints/GPM code

### ISSUE THREAD

**Issue by fmessmer** [4am-robotics/cob_control#152]
- [ ] remove priority template: `<typename PRIO = uint32_t>` @ipa-fxm-cm 
 - [ ] separate and clearly name constraint builder/factory from actual constraints
 - [ ] use *.cpp and create library instead of *_impl.h
 - [ ] extract abstract GPM solver (i.e. just GPM formular) from actual GPM implementation (i.e. sum over constraint list)
 - [ ] allow to use constraints with/without prediction, toggleable via parameter (could help with #64)

---

 - [ ] optional: remove dynamic reconfigurability for solver types @ipa-fxm-cm

@fmessmer FYI

**Comment by fmessmer** [comment]
@ipa-fxm-cm 
I think we should keep the dynamic reconfigurability for the time being...so could you please focus on "remove priority template" first?

**Comment by bbrito** [comment]
I agree. At the moment for me at least dynamic reconfigure is really useful. But once we finish all this solvers and make them stable we can start removing it.

**Comment by fmessmer** [comment]
In https://github.com/ipa320/cob_control/pull/126 we started to debug #125, but did not pursue this any further....thus it has been closed unmerged...

**Comment by fmessmer** [comment]
getting rid of the priority template is not that easy....I started it in [my branch](https://github.com/ipa-fxm/cob_control/tree/remove_prio_template), but I'm not able to compile it yet...


Problem is:
When trying to get rid of the priority template, the whole [PriorityBase class](https://github.com/ipa320/cob_control/blob/indigo_dev/cob_twist_controller/include/cob_twist_controller/constraints/constraint_base.h#L51) actually is obsolete and could be removed leaving [ConstraintBase](https://github.com/ipa320/cob_control/blob/indigo_dev/cob_twist_controller/include/cob_twist_controller/constraints/constraint_base.h#L124) as the top abstract class for constraints...
However, I then ran into issues with a templated typedef for [ConstraintBase_t](https://github.com/ipa320/cob_control/blob/indigo_dev/cob_twist_controller/include/cob_twist_controller/constraints/constraint_base.h#L267) which is used all over the place...and I couldn't resolve it yet...
